Source/MQTTnet/Implementations/MqttTcpServerListener.cs

```diff
@@ -178,11 +178,24 @@ async Task TryHandleClientConnectionAsync(CrossPlatformSocket clientSocket)
                 {
                     var sslStream = new SslStream(stream, false, _tlsOptions.RemoteCertificateValidationCallback);
 
-                    await sslStream.AuthenticateAsServerAsync(
-                        _tlsCertificate,
-                        _tlsOptions.ClientCertificateRequired,
-                        _tlsOptions.SslProtocol,
-                        _tlsOptions.CheckCertificateRevocation).ConfigureAwait(false);
+                    #if NETCOREAPP3_1 || NET5_0_OR_GREATER
+                        await sslStream.AuthenticateAsServerAsync(
+                            new SslServerAuthenticationOptions()
+                            {
+                                ServerCertificate = _tlsCertificate,
+                                ClientCertificateRequired = _tlsOptions.ClientCertificateRequired,
+                                EnabledSslProtocols = _tlsOptions.SslProtocol,
+                                CertificateRevocationCheckMode = _tlsOptions.CheckCertificateRevocation ? X509RevocationMode.Online : X509RevocationMode.NoCheck,
+                                EncryptionPolicy = EncryptionPolicy.RequireEncryption,
+                                CipherSuitesPolicy = _tlsOptions.CipherSuitesPolicy
+                            });
```

Comment:
Please add .ConfigureAwait...

Reply:
Oh yes, You are totally right. I have added it.